lib/Ohmbrewer_RIMS.cpp

```diff
@@ -4,120 +4,89 @@
+    int tempBus;
+    if ( (size == 2) || (size == 3) ){
+        tempBus = thermPins->front();
+        _tube = new Thermostat(id+3, thermPins);
+        _tunSensor = new TemperatureSensor(id+1, tempBus);
     }else{
         //publish error
         Ohmbrewer::Publisher::publish_map_t pMap;
         Ohmbrewer::Publisher* pub = new Ohmbrewer::Publisher(new String("error_log"), &pMap);
-
-        pMap[String("array_check_rims")] = String("improperly formed array - RIMS(int, int[], int, int[], int, bool, String)");
+        pMap[String("list_check_rims")] = String("improperly formed array - RIMS(int, int<list>, int");
         pub->publish();
     }
 
+    _recirc = new Pump(id+2, pumpPin);
     registerUpdateFunction();
 }
 
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Equipment
- * @param tubePins[ temp busPin ; heating powerPin ; heating controlPin ]
- * @param tunBus - mash tun temperature pin
- * @param pumpPins[powerPin ; controlPin ]
+ * @param thermPins list with formatting of: [ temp busPin ;  heating controlPin ; heating powerPin ]
+ * @param pumpPin - Single speed pump will only have PowerPin
  * @param stopTime The time at which the Equipment should shut off, assuming it isn't otherwise interrupted
  * @param state Whether the Equipment is ON (or OFF). True => ON, False => OFF
  * @param currentTask The unique identifier of the task that the Equipment believes it should be processing
  * @param targetTemp The new target temperature in Celsius
  */
-Ohmbrewer::RIMS::RIMS(int id, int (&tubePins)[3], int tunBus, int (&pumpPins)[2], int stopTime,
+Ohmbrewer::RIMS::RIMS(int id, std::list<int>* thermPins, int pumpPin, int stopTime,
                       bool state, String currentTask, const double targetTemp) : Ohmbrewer::Equipment(id, stopTime, state, currentTask) {
-
-    _tube = new Thermostat(id+2, tubePins, targetTemp);
-    _tunSensor = new TemperatureSensor(id+1, tunBus);
-    int n = sizeof(pumpPins) / sizeof(int);
-    Ohmbrewer::Publisher::publish_map_t pMap;
-    Ohmbrewer::Publisher* pub = new Ohmbrewer::Publisher(new String("error_log"), &pMap);
-    if ( n > 1){
-        _recirc = new Pump(1, pumpPins[0], pumpPins[1]);
-    }else if (n == 1){
-        _recirc = new Pump(1, pumpPins[0], -1);//Single speed pump
+    int size = thermPins->size();
+    int tempBus;
+    if ( (size == 2) || (size == 3) ){
+        tempBus = thermPins->front();
+        _tube = new Thermostat(id+3, thermPins, targetTemp);
+        _tunSensor = new TemperatureSensor(id+1, tempBus);
     }else{
         //publish error
         Ohmbrewer::Publisher::publish_map_t pMap;
         Ohmbrewer::Publisher* pub = new Ohmbrewer::Publisher(new String("error_log"), &pMap);
-
-        pMap[String("array_check_rims")] = String("improperly formed array - RIMS(int , int[], int, int[], int, bool, String, double)");
+        pMap[String("list_check_rims")] = String("improperly formed array - RIMS(int, int<list>, int");
```

Comment:
Side note: We should make another Publisher constructor that takes a single string such that you can do lines 84-85 in a single line. That seem slike it would be handy.